modules/vpc/main.tf

```diff
@@ -0,0 +1,196 @@
+locals {
+  azs                = slice(data.aws_availability_zones.available.names, 0, 3)
+  private_cidr_range = [for k, v in local.azs : cidrsubnet(var.vpc_cidr, 8, k)]
+  public_cidr_range  = [for k, v in local.azs : cidrsubnet(var.vpc_cidr, 8, k + 4)]
+}
+
+resource "aws_vpc" "new_vpc" {
+  cidr_block           = var.vpc_cidr
+  enable_dns_hostnames = true
+  enable_dns_support   = true
+
+  tags = merge(
+    {
+      "Name" = "${var.name_prefix}-vpc"
+    },
+    var.tags,
+  )
+}
+
+resource "aws_vpc_endpoint" "s3" {
+  vpc_id       = aws_vpc.new_vpc.id
+  service_name = "com.amazonaws.${data.aws_region.current.name}.s3"
+  route_table_ids = concat(
+    aws_route_table.private_routes.*.id,
+    aws_route_table.default.*.id,
+  )
+
+  tags = var.tags
+}
+
+resource "aws_vpc_dhcp_options" "main" {
+  domain_name         = data.aws_region.current.name == "us-east-1" ? "ec2.internal" : format("%s.compute.internal", data.aws_region.current.name)
+  domain_name_servers = ["AmazonProvidedDNS"]
+
+  tags = var.tags
+}
+
+resource "aws_vpc_dhcp_options_association" "main" {
+  vpc_id          = aws_vpc.new_vpc.id
+  dhcp_options_id = aws_vpc_dhcp_options.main.id
+}
+
+resource "aws_internet_gateway" "igw" {
+  vpc_id = aws_vpc.new_vpc.id
+
+  tags = merge(
+    {
+      "Name" = "${var.name_prefix}-igw"
+    },
+    var.tags,
+  )
+}
+
+resource "aws_route_table" "default" {
+  vpc_id = aws_vpc.new_vpc.id
+
+  tags = merge(
+    {
+      "Name" = "${var.name_prefix}-public"
+    },
+    var.tags,
+  )
+}
+
+resource "aws_main_route_table_association" "main_vpc_routes" {
+  vpc_id         = aws_vpc.new_vpc.id
+  route_table_id = aws_route_table.default.id
+}
+
+resource "aws_route" "igw_route" {
+  destination_cidr_block = "0.0.0.0/0"
+  route_table_id         = aws_route_table.default.id
+  gateway_id             = aws_internet_gateway.igw.id
+
+  timeouts {
+    create = "20m"
```

Comment:
maybe it can be a variable with a default value

Reply:
Lets push this in another PR